### PR TITLE
Added CRUD operations and marshmallow schema for reader category entity

### PR DESCRIPTION
**Pull request by mBerezovskyy**

### CODE REVIEW

File: controllers/reader_categories_controller.py [hunk 1]
```diff
@@ -2,68 +2,68 @@
...
 
 
 @reader_categories_bp.route("/", methods=["POST"])
 def create_reader_category():
-    new_reader_category = request.get_json()
-    new_reader_category.id = 6
-    return jsonify(new_reader_category)
+    body = request.get_json()
+    category = body.get('category')
+    reader_category_already_exists = ReaderCategory.query.filter_by(category=category).first()
```
Comment (Ari100telll) [inline controllers/reader_categories_controller.py]: Check is SQLAlchemy have somthing like .exists()

**Review by KhrystynaVysotska — DISMISSED** [review]

**Review by Ari100telll — DISMISSED** [review]

**Review by Ravenen — DISMISSED** [review]

File: controllers/users_controller.py [hunk 2]
```diff
@@ -2,90 +2,61 @@
...
-    return jsonify(updated_user)
+    body = request.get_json()
+
+    user_update_body = {
+        "name": body.get("name", None),
+        "surname": body.get("surname", None),
+        "phone_number": body.get("phone_number", None),
+        "address": body.get("address", None),
+        "reader_category_id": body.get("reader_category_id", None)
+    }
```
Comment (Ravenen) [inline controllers/users_controller.py:46]: Why don't you use `from_dict` method of the user model?

File: controllers/rents_controller.py [hunk 3]
```diff
@@ -2,212 +2,60 @@
...
+    rent_update_body = {
+        "library_user_id": body.get("library_user_id", None),
+        "rent_start_date": body.get("rent_start_date", None),
+        "expected_rent_end_date": body.get("expected_rent_end_date", None),
+        "rent_end_date": body.get("rent_end_date", None),
+        "library_item_id": body.get("library_item_id", None),
+        "rent_price": body.get("rent_price", None),
+        "fine_price": body.get("fine_price", None),
+        "damage_level_id": body.get("damage_level_id", None)
+    }
```
Comment (Ravenen) [inline controllers/rents_controller.py:48]: Why don't you use `from_dict`?

File: controllers/damage_levels_controller.py [hunk 4]
```diff
@@ -9,61 +17,40 @@
...
 def update_damage_level(damage_level_id: int):
-    updated_damage_level = request.get_json()
-    updated_damage_level["id"] = damage_level_id
-    return jsonify(updated_damage_level)
+    body = request.get_json()
+
+    damage_level_update_body = {
+        "level": body.get("level", None),
+        "fine_percentage": body.get("fine_percentage", None),
+    }
```
Comment (Ravenen) [inline controllers/damage_levels_controller.py:46]: Why don't you use `from_dict`?

**Review by Ari100telll — APPROVED** [review]

**Review by Ravenen — APPROVED** [review]

**Pull request merged**